File: app/src/main/java/com/example/gallery/AlbumRecyclerViewAdapter.java
package com.example.gallery;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.gallery.databinding.AlbumViewBinding;
import com.example.gallery.databinding.PhotoViewBinding;

public class AlbumRecyclerViewAdapter extends RecyclerView.Adapter<AlbumRecyclerViewAdapter.MyViewHolder> {

    private int imageId[];
    private String description[];

    public AlbumRecyclerViewAdapter(int img[], String des[]) {
        imageId = img;
        description = des;
    }
    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        AlbumViewBinding binding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.getContext()),
                R.layout.album_view, parent, false);

        return new MyViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.binding.ivPhotoAlbum.setImageResource(imageId[position]);
        holder.binding.tvTextAlbum.setText(description[position]);
    }

    @Override
    public int getItemCount() {
        return imageId.length;
    }

    public static final class MyViewHolder extends RecyclerView.ViewHolder {
        private final AlbumViewBinding binding;
        public MyViewHolder(@NonNull AlbumViewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/example/gallery/AlbumRecyclerViewAdapter.kt
package com.example.gallery

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.gallery.databinding.AlbumViewBinding

class AlbumRecyclerViewAdapter(
    private val list: List<Pair<String, List<String>>>,
    private val albumListener: AlbumListener,
) : RecyclerView.Adapter<AlbumRecyclerViewAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val binding: AlbumViewBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.album_view, parent, false
        )
        return MyViewHolder(binding, albumListener)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.bind(list[position])
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class MyViewHolder(
        val binding: AlbumViewBinding,
        private val albumListener: AlbumListener,
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: Pair<String, List<String>>) {
            val (title, list) = item

            Glide.with(itemView)
                .load(list.first())
                .into(binding.ivPhotoAlbum)

            binding.tvTextAlbum.text = albumText(title)

            binding.root.setOnClickListener {
                albumListener.onAlbumClick(list)
            }
        }

        private fun albumText(str: String): String {
            var txt = ""
            var i = str.length-1
            while (i >= 0) {
                if (!(str[i].equals('/'))) {
                    txt += str[i]
                } else {
                    i = -1
                }
                i--
            }
            return txt.reversed()
        }
    }

    interface AlbumListener {
        fun onAlbumClick(list: List<String>)
    }
}